src/benchmark/java/com/eatthepath/UUIDBenchmark.java

```diff
@@ -54,21 +55,38 @@ public void setup() {
 
     @Benchmark
     public UUID benchmarkUUIDFromString() {
-        return UUID.fromString(this.uuidStrings[this.i++ % PREGENERATED_UUID_COUNT]);
+        resetCounterIfNecessary();
+        return UUID.fromString(this.uuidStrings[(int) (this.i++ % PREGENERATED_UUID_COUNT)]);
     }
 
     @Benchmark
     public UUID benchmarkUUIDFromFastParser() {
-        return FastUUID.parseUUID(this.uuidStrings[this.i++ % PREGENERATED_UUID_COUNT]);
+        resetCounterIfNecessary();
+        return FastUUID.parseUUID(this.uuidStrings[(int) (this.i++ % PREGENERATED_UUID_COUNT)]);
+    }
+
+    // Checking if type-check to String won't affect performance
+    @Benchmark
+    public UUID benchmarkUUIDFromCharSequenceFastParser() {
+        resetCounterIfNecessary();
+        return FastUUID.parseUUID((CharSequence) this.uuidStrings[(int) (this.i++ % PREGENERATED_UUID_COUNT)]);
     }
 
     @Benchmark
     public String benchmarkUUIDToString() {
-        return this.uuids[this.i++ % PREGENERATED_UUID_COUNT].toString();
+        resetCounterIfNecessary();
+        return this.uuids[(int) (this.i++ % PREGENERATED_UUID_COUNT)].toString();
     }
 
     @Benchmark
     public String benchmarkFastParserToString() {
-        return FastUUID.toString(this.uuids[this.i++ % PREGENERATED_UUID_COUNT]);
+        resetCounterIfNecessary();
+        return FastUUID.toString(this.uuids[(int) (this.i++ % PREGENERATED_UUID_COUNT)]);
+    }
+
+    private void resetCounterIfNecessary() {
+        if (this.i == Long.MAX_VALUE) {
+            this.i = 0;
+        }
```

Comment:
This might not be necessary, but _better safe than sorry_ 🙂 
We don't want tests to run on future amazing hardware ;-)

But sure, we can do that.

Ideally, though, each benchmark should have its own state object instead of making state be shared.